fs/fs.go

```diff
@@ -295,3 +325,83 @@ func getVfsStats(path string) (uint64, uint64, uint64, error) {
 	avail := uint64(s.Frsize) * s.Bavail
 	return total, free, avail, nil
 }
+
+func dockerStatusValue(status [][]string, target string) string {
+	for _, v := range status {
+		if len(v) == 2 && strings.ToLower(v[0]) == strings.ToLower(target) {
+			return v[1]
+		}
+	}
+	return ""
+}
+
+func dockerDMDevice(driverStatus string) (string, uint, uint, error) {
+	var config [][]string
+	err := json.Unmarshal([]byte(driverStatus), &config)
+	if err != nil {
+		return "", 0, 0, err
+	}
+	poolName := dockerStatusValue(config, "Pool Name")
+	if len(poolName) == 0 {
+		return "", 0, 0, fmt.Errorf("Could not get dm pool name")
+	}
+
+	poolName = path.Join("/dev/mapper", poolName)
+
+	dmTable, err := exec.Command("dmsetup", "table", poolName).Output()
+	if err != nil {
+		return "", 0, 0, err
+	}
+
+	var (
+		major, minor, bkt uint
+		bkts              string
+	)
+
+	_, err = fmt.Fscanf(bytes.NewReader(dmTable),
+		"%d %d %s %d:%d %d:%d %d %d %d %s",
+		&bkt, &bkt, &bkts, &bkt, &bkt, &major, &minor, &bkt, &bkt, &bkt, &bkts)
+	if err != nil {
+		return "", 0, 0, err
+	}
+	return poolName, major, minor, nil
+}
+
+func getDMStats(poolName string) (uint64, uint64, uint64, error) {
+	dmTable, err := exec.Command("dmsetup", "table", poolName).Output()
+	if err != nil {
+		return 0, 0, 0, err
+	}
+
+	var (
+		dataBlkSize, bkt uint64
+		bkts             string
+	)
+
+	_, err = fmt.Fscanf(bytes.NewReader(dmTable),
+		"%d %d %s %d:%d %d:%d %d %d %d %s",
+		&bkt, &bkt, &bkts, &bkt, &bkt, &bkt, &bkt, &dataBlkSize, &bkt, &bkt, &bkts)
```

Comment:
Should we cache the value of `dataBlkSize` from the previous call to `dmsetup table <poolName>` instead?